packages/core/src/browser/quick-open/quick-open-service.ts

```diff
@@ -17,6 +17,12 @@
 import { injectable } from 'inversify';
 import { QuickOpenModel } from './quick-open-model';
 import { MessageType } from '../../common/message-service-protocol';
+import URI from 'vscode-uri';
+
+export interface TitleButton {
```

Comment:
- it should be named `QuickInputTitleButton`, please always use specific names
- it should not belong here, but to `quick-input-service`
- core should not know anything about `iconPath`, it should be just `icon: string` which points to a css class
- `iconPath` should be properly turned into `iconUrl` by the plugin system, see how it is done for tree view icons
  - plugin host part: https://github.com/theia-ide/theia/blob/8c0d234e9c04d9ee37157dc6f66e31645d0de300/packages/plugin-ext/src/plugin/tree/tree-views.ts#L226-L255
  - main part: https://github.com/theia-ide/theia/blob/26492e7d6af261be800be6c91d43be39256d59f8/packages/plugin-ext/src/main/browser/view/tree-views-main.tsx#L196-L210